modules/mbedtls/Kconfig.mbedtls

```
@@ -46,28 +46,33 @@ comment "Supported key exchange modes"
 
 config MBEDTLS_RSA_C
 	bool "RSA base support"
+	default y if UOSCORE || UEDHOC
 
 if MBEDTLS_RSA_C
 
 config MBEDTLS_PKCS1_V15
 	bool "RSA PKCS1 v1.5"
+	default y if UOSCORE || UEDHOC
 
 config MBEDTLS_PKCS1_V21
 	bool "RSA PKCS1 v2.1"
+	default y if UOSCORE || UEDHOC
 
 config MBEDTLS_GENPRIME_ENABLED
 	bool "Prime number generation code"
 
 endif # MBEDTLS_RSA_C
 
 config MBEDTLS_RSA_FULL
-	bool
-	select MBEDTLS_RSA_C
-	select MBEDTLS_PKCS1_V15
-	select MBEDTLS_PKCS1_V21
+	def_bool y
+	depends on MBEDTLS_RSA_C && MBEDTLS_PKCS1_V15 && MBEDTLS_PKCS1_V21
```

Based on [this comment](https://github.com/zephyrproject-rtos/zephyr/pull/89200/files#r2073105170) I don´t know if `MBEDTLS_RSA_FULL` is what we're looking for. It was OK when it was selecting all the other RSA related features, but now it makes less sense IMO. What about renaming `MBEDTLS_RSA_FULL` to `MBEDTLS_RSA_SOME` and change the `depends on` as follows:
```suggestion
	depends on MBEDTLS_PKCS1_V15 || MBEDTLS_PKCS1_V21
```
This is based on the fact that  `MBEDTLS_PKCS1_V15` and `MBEDTLS_PKCS1_V21` are already gated by `MBEDTLS_RSA_C`.